transport/src/test/java/io/netty/channel/pool/FixedChannelPoolTest.java

```diff
@@ -297,6 +305,12 @@ public void run() {
         pool.release(channel).syncUninterruptibly();
         sc.close().syncUninterruptibly();
         channel.close().syncUninterruptibly();
+
+        ChannelPool.Stats stats = pool.stats();
+        assertEquals(0, stats.leasedChannelsCount());
+        assertEquals(0, stats.pendingChannelsCount());
+        // TODO: (need to wait the close operation is completed. how?
+//        assertEquals(0, stats.availableChannelsCount());
```

Comment:
@trustin because that would break the interface and we could not anymore extend "AutoClosable". I think it was just a bad idea to not make it async in the first place :( Maybe our best bet is in the long run (as you already said in another issue partly) to `@deprecated` our pool stuff and create a new interface + impls etc.